dali/c_api/c_api_test.cc

```diff
@@ -472,4 +472,47 @@ TYPED_TEST(CApiTest, TestExecutorMeta) {
   daliFreeExecutorMetadata(meta, N);
 }
 
+TYPED_TEST(CApiTest, ExternalSourcePinnedMemoryUseCopyKernel) {
+  TensorListShape<> input_shape = {{37, 23, 3}, {12, 22, 3}, {42, 42, 3}, {8, 8, 3},
+                                   {64, 32, 3}, {32, 64, 3}, {20, 20, 3}, {64, 64, 3},
+                                   {10, 10, 3}, {60, 50, 3}, {10, 15, 3}, {48, 48, 3}};
+  TensorList<CPUBackend> input_cpu;
```

Comment:
Just for the safety I would explicitly set input_cpu as pinned, just in case we will make Tensors not pinned by default.

Reply:
The one that should be pinned is `input`. `input_cpu` is then copied to `input` that can be CPU or GPU. I'll set it to pinned

Reply:
done